azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -319,27 +320,67 @@ public void updateProjectDescription(final Project project, final String descrip
     return this.projectLoader.getProjectEvents(project, results, skip);
   }
 
-  public Props getProperties(final Project project, final String source)
-      throws ProjectManagerException {
-    return this.projectLoader.fetchProjectProperty(project, source);
+  public Props getPropertiesFromFlowFile(final Flow flow, final String jobName, final String source,
```

Comment:
Agree. 
Here source is the flow file name. I think it also makes sense to have flow file name as the argument. Probably I can change the argument name to flowFileName.

Reply:
It could be more reusable. But if we extract project id, project version, flow id, flow version from flow object, then the argument list would become relatively long (6 arguments in total), which might not be a good coding practice.
`When a function seems to need more than two or three arguments, it is likely that some of those arguments ought to be wrapped into a class of their own.` (Excerpt From: Martin, Robert C. “Clean Code: A Handbook of Agile Software Craftsmanship.” iBooks.)